### PR TITLE
Updates to travis scripts

### PR DESCRIPTION
**Pull request by spookydonut**
checks templates now and some other things

### CODE REVIEW

File: tools/travis/template_dm_generator.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python
+
+import os
+import os.path
```
Comment (psykzz) [inline tools/travis/template_dm_generator.py]: redundant line

Comment (spookydonut) [reply tools/travis/template_dm_generator.py]: same as tg

Comment (Rohesie) [reply tools/travis/template_dm_generator.py]: Seems to be redundant from what it's said, just accept the suggestion.

**Comment by Rohesie** [comment]
You'll have to remove the icon state generation check because the only cure for that right now would be a nuke.

File: tools/travis/template_dm_generator.py (hunk 1 above)
Comment (spookydonut) [reply tools/travis/template_dm_generator.py]: > 
> 
> Seems to be redundant from what it's said, just accept the suggestion.

i just mean i'll change it upstream

**Pull request merged**